src/tasks/WorkloadBuildTasks/PackageInstaller.cs

```diff
@@ -19,29 +19,31 @@ internal sealed class PackageInstaller
         private string _nugetConfigContents;
         private TaskLoggingHelper _logger;
         private string _packagesDir;
+        private bool _cleanPackagesdir;
```

Comment:
should this be true by default ?

Reply:
This is now set `_cleanPackagesdir = packagesPath is null;` in the constructor. So, if `packagesPath` is set, IOW, we are using a shared packages path then we don't want it to be cleaned.
Also, the main task now creates all the temp files under a top level temporary directory which gets removed at the end. And that would include these paths too.

Reply:
Very cool. Would it make sense to not delete the directory when running on local machine && something failed ?

Reply:
I guess `SKIP_PROJECT_CLEANUP=1` still works ?

Reply:
Good idea. I was disabling it in code when needed! I can add a similar env var for this. SKIP_PROJECT_CLEANUP is for WBT.

Reply:
I'll include this in a follow up PR.